.github/workflows/handler-pr-close.yml

```diff
@@ -83,27 +88,88 @@ jobs:
             workspace    = "$TFE_WORKSPACE"
           }
           EOF
-    # this step will get the workspace created for google-private-tcp-active-active for this PR
-      - name: Get workspace for google-private-tcp-active-active
-        id: get-workspaces-google-private-tcp-active-active
+
+    # this step will check if workspace is created for google-private-tcp-active-active for this PR
+      - name: Check if workspace for google-private-tcp-active-active exists
+        id: check-workspaces-google-private-tcp-active-active
         working-directory: ${{ env.WORK_DIR_PATH }}
         run: >-
-          echo "::set-output name=WORSPACE_TCP_AA::$(terraform workspace list -input=false -no-color | grep google-private-tcp-active-active-${{ github.event.pull_request.number }})"
+          echo "::set-output name=WORSPACE_TCP_AA_EXISTS::$(curl -s \  --header "Authorization: Bearer $TFE_TOKEN" \  --header "Content-Type: application/vnd.api+json" \  https://app.terraform.io/api/v2/organizations/$TFE_ORGANIZATION/workspaces | jq -r '.data[] | .attributes | .name' | grep -cx google-private-tcp-active-active-${{ github.event.pull_request.number }})"
```

Comment:
The [List Workspaces](https://www.terraform.io/cloud-docs/api-docs/workspaces#list-workspaces) API is paginated so you could get a false result here if the workspace you are looking for exists but isn't returns on the first page of results.

It would be better to use the [Show Workspace](https://www.terraform.io/cloud-docs/api-docs/workspaces#show-workspace) API, specifically the route that shows a workspace by name (`GET /organizations/:organization_name/workspaces/:name`) to check if the workspace already exists.

Also, I don't believe you need the backslash characters here since this is on a single line.

Reply:
@sudomateo , Thanks for your review. Replaced [List Workspaces](https://www.terraform.io/cloud-docs/api-docs/workspaces#list-workspaces) API with [Show Workspace](https://www.terraform.io/cloud-docs/api-docs/workspaces#show-workspace) API and removed the backslash characters.
Also made env var `PR_NUMBER: ${{ github.event.pull_request.number }}` to reduce the length of URL.